### PR TITLE
add todo (need review)

### PR DESCRIPTION
**Pull request by adrienluitot**

### CODE REVIEW

File: Todo.py
Comment (s0lst1ce) [inline Todo.py]: @adrienluitot you should delete this token

File: Todo.py
Comment (s0lst1ce) [inline Todo.py]: you don't need to do this in a cog, not hte way the bot is built. So you can remove the listener.

File: Todo.py
Comment (s0lst1ce) [inline Todo.py]: is it a channel or a file ? This seems unclear to me...

Comment (s0lst1ce) [reply Todo.py]: Also where do you define this variable ?

File: Todo.py
Comment (s0lst1ce) [inline Todo.py]: you only need the first line so instead of reading the whole file you can just do `readline` which will only read one line with each iteration.

File: Todo.py
Comment (s0lst1ce) [inline Todo.py]: instead of having a single allowed name "Public repost" have global var so that is can be configured once I implement #4

File: Todo.py
Comment (s0lst1ce) [inline Todo.py:51]: shouldn't the use of this group of commands be restricted ?

File: Todo.py
Comment (adrienluitot) [reply Todo.py]: It's an error from, I corrected that, It's a file I also fixed that

File: Todo.py
Comment (adrienluitot) [reply Todo.py]: Thanks, will fix that

File: Todo.py
Comment (adrienluitot) [reply Todo.py]: What's the goal ?

File: Todo.py
Comment (adrienluitot) [reply Todo.py:51]: Yes ! It should :D

File: Todo.py
Comment (s0lst1ce) [reply Todo.py]: one day the bot might extend its reach to several servers so I want to keep it ready for such an eventuality.

File: Todo.py
Comment (s0lst1ce) [reply Todo.py]: Ok I'll wait for the commit then ;)

File: Todo.py
Comment (NotaSmartDev) [reply Todo.py]: This line should be deleted before I can merge...

File: Todo.py
Comment (s0lst1ce) [inline Todo.py]: Where is `TODO_TYPES` defined at ? Your PR doesn't include the settings.py file so I'm guessing you didn't change it ? I'm a bit lost tbh.

File: Todo.py
Comment (s0lst1ce) [reply Todo.py]: This line should be deleted before I can merge...

File: Todo.py
Comment (s0lst1ce) [reply Todo.py]: Same here, as you agreed I need you to delete this listener.

File: Todo.py
Comment (s0lst1ce) [reply Todo.py]: As we now know that the bot is going to be used in more than one server, I'll  be waiting for you to fix this.

File: Todo.py
Comment (s0lst1ce) [reply Todo.py:51]: Maybe you should add a new clearance level that could be configured ( see #4 ) ?

File: Todo.py
Comment (s0lst1ce) [inline Todo.py]: camel case shouldn't be used in python. Ever. If you'r referencing classes you can do something like `ChannelName`. Functions and variables should look like: `channel_name`. See PEP8 for more info.

File: Todo.py
Comment (s0lst1ce) [inline Todo.py:70]: What is the point of the space in `" {}"` ?

File: Todo.py
Comment (s0lst1ce) [inline Todo.py:71]: Maybe letting users issue commands with multiple args only using spaces would make it more natural to them. This would make it closer to how the CLI works as well as most other bots on discord. It also wouldn't cause any additional difficulty and would probably be clearer.

File: Todo.py
Comment (s0lst1ce) [inline Todo.py]: Again, camel case

File: Todo.py
Comment (s0lst1ce) [inline Todo.py]: Again, camel case

File: Todo.py
Comment (s0lst1ce) [inline Todo.py:76]: I think I already mentioned it but as I'm seeing it again: I think using `readline` would be more efficient.

File: Todo.py
Comment (s0lst1ce) [inline Todo.py:92]: I used to cut the string this way too but it's most unefficient. Look at my updated code instead. If you don't see where I used to do this and how I solved this please ping me and I'll do my best to show you how to use the API to its full extent.

File: Todo.py
Comment (s0lst1ce) [inline Todo.py:93]: Maybe you could use a `discord.py` converter here ? Not sure if it'd be better, I'll let you judge about it but keep me informed ;)

File: Todo.py
Comment (s0lst1ce) [inline Todo.py]: Again, camel case
```suggestion
                        repost_channel = chan
```

File: Todo.py
Comment (s0lst1ce) [inline Todo.py]: Again, camel case

File: Todo.py
Comment (s0lst1ce) [inline Todo.py:101]: Again, you don't need to cut your strings in pieces like this.

File: Todo.py
Comment (s0lst1ce) [inline Todo.py]: Again, camel case. And I'll stop writing this every time I see camel case again. I'll let you review you own code ;)

File: Todo.py
Comment (s0lst1ce) [inline Todo.py:130]: Should we add more "natural" names for the colors ? This could be useful not only for this project but for other commands too. May even save us, the coders some time ^^

File: Todo.py
Comment (s0lst1ce) [inline Todo.py]: It is recommended to add a newline at the end of a file.
```suggestion
    bot.add_cog(Todo(bot))
    
```

**Review by s0lst1ce — CHANGES_REQUESTED** [review]
A few minor modifications should be commited before I can merge this PR into `rewrite`. You've done a great job so I think you'll be able to solve all of those issues by yourself but if you do need/want help don't hesitate to ping me. I'll try and be available although I don't code so much :(

File: Todo.py
Comment (adrienluitot) [reply Todo.py]: Already done ;)

File: Todo.py
Comment (adrienluitot) [reply Todo.py]: I didn't psh yet but done

File: Todo.py
Comment (adrienluitot) [reply Todo.py]: I made a PUBLIC_REPOST constant, even if I'm not sure if it was what you wanetd

File: Todo.py
Comment (s0lst1ce) [reply Todo.py]: Why am I not seeing it then ? You haven't you committed ?

File: Todo.py
Comment (adrienluitot) [reply Todo.py]: I didn't push the changes but I added it (in settings.py)

File: Todo.py
Comment (s0lst1ce) [reply Todo.py]: Oh I see. Well please commit asap. It would prevent me from writing useless comments ;)

File: Todo.py
Comment (adrienluitot) [reply Todo.py]: Oh ! okay I confused camel case and snake case >.< sorry !

File: Todo.py
Comment (adrienluitot) [reply Todo.py:70]: In fact I think I had copy your code x) and It's probably to keep spaces `(to keep "make a todo list" instead of "makeatodolist")` :smile:

File: Todo.py
Comment (adrienluitot) [reply Todo.py:71]: Hum, the problem is that I need to keep spaces for todo's descriptions. Otherwise they have to use quotes.

File: Todo.py
Comment (adrienluitot) [reply Todo.py]: Changed it ;)

File: Todo.py
Comment (adrienluitot) [reply Todo.py:76]: Yep, done but I didn't push yet

File: Todo.py
Comment (adrienluitot) [reply Todo.py:92]: Okay I'll take a look

File: Todo.py
Comment (s0lst1ce) [reply Todo.py]: :man_facepalming: Sry I'm such an idiot. I read through your code a little too fast it would seem. I completely missed misunderstood the use of this thing. It probably wasn't needed. Sry I made a big mistake here :grimacing:

File: Todo.py
Comment (adrienluitot) [reply Todo.py:92]: Is this only `member:discord.Member` in the parameters of the function ? Else I don't see what you done

File: Todo.py
Comment (adrienluitot) [reply Todo.py:93]: How do you do ? ^^

File: Todo.py
Comment (adrienluitot) [reply Todo.py]: Already done in my code, sorry to avoid problem I don't commit what you did :open_mouth:

File: Todo.py
Comment (adrienluitot) [reply Todo.py:101]: How would you do so ? :smiley:

File: Todo.py
Comment (adrienluitot) [reply Todo.py:130]: Hmm yeah not bad idea but, then how ? users will use text like "red" ? 
Also there is some basic colors already done see https://discordpy.readthedocs.io/en/latest/api.html#discord.Colour

File: Todo.py
Comment (adrienluitot) [reply Todo.py]: Same as your last change :D

File: Todo.py
Comment (adrienluitot) [reply Todo.py]: it's that

File: Todo.py
Comment (adrienluitot) [reply Todo.py]: Oh, so I remove it ? We can also let it in memory to make translation xD

Comment (s0lst1ce) [reply Todo.py]: Do whatever you want I don't have my word in this matter anymore...

File: Todo.py
Comment (s0lst1ce) [reply Todo.py]: np

File: Todo.py
Comment (s0lst1ce) [reply Todo.py:70]: Ok

File: Todo.py
Comment (s0lst1ce) [reply Todo.py:71]: That is true. Although a way to fix this would be to put the message at the end. Like this all words after the commands would be part of the message. What do you think ?

File: Todo.py
Comment (s0lst1ce) [reply Todo.py:92]: I checked and that seems to be it. I didn't want to say that was it for sometimes I forget part of my code ^^

File: Todo.py
Comment (s0lst1ce) [reply Todo.py:93]: Look at the doc :P

File: Todo.py
Comment (s0lst1ce) [reply Todo.py:101]: Again using a converter.

File: Todo.py
Comment (s0lst1ce) [reply Todo.py:130]: oh I didn't know. Well this may reveal itself useful one day

**Review by s0lst1ce — APPROVED** [review]

**Comment by s0lst1ce** [comment]
I'm going to merge your PR and then you'll be able to switch to converters after-wise but there's no reason to withhold the merge into `rewrite` now.

**Pull request merged**
